docs/data-import.rst

```diff
@@ -43,49 +53,78 @@ Run the import command:
     python manage.py import_nc --dest $PWD/ncdata
 
 This took ~25 minutes on my laptop. Run ``tail -f traffic_stops.log`` to follow
-along.
+along.  Reusing an existing ``--dest`` directory will speed up import.  However,
+if import code has changed since the last time the directory was used, don't
+reuse an existing directory.
 
 Now you should be able to view data with ``runserver``:
 
 .. code-block:: bash
 
     python manage.py runserver
 
+Admin
++++++
+
+If the state database doesn't exist or migrations haven't been applied to it,
+fix that first.
+
+Access /admin/tsdata/dataset/ and create a "dataset" describing the data to be
+imported.  The normal MD URL is stored in ``md.data.__init__.py``; the normal
+NC URL is stored in ``nc.management.commands.import_nc.py``.  Newer datasets
+may be available at https://s3-us-west-2.amazonaws.com/openpolicingdata/.  Be
+sure to specify a destination directory.
+
+Once the "dataset" has been created, select the new dataset in list view and
+apply the "Import selected dataset" action.
 
 Server
 ------
 
-Raw NC Data
-___________
+The PostgreSQL user must have SUPERUSER privileges to perform the import.
+Depending on current admin policies, that may have to be granted and
+revoked around the import.
 
-To start with fresh NC data, first drop the server's database:
+Temporarily grant our PostgreSQL user SUPERUSER privileges:
 
 .. code-block:: bash
 
-    sudo -u postgres dropdb traffic_stops_nc_staging
+    sudo -u postgres psql -c 'ALTER USER traffic_stops_staging WITH SUPERUSER;'
 
-Then run a **deploy** to recreate the database.
-
-Temporarily grant our PostgreSQL user SUPERUSER privileges:
+When finished, revoke SUPERUSER privileges:
 
 .. code-block:: bash
 
-    sudo -u postgres psql -c 'ALTER USER traffic_stops_production WITH SUPERUSER;'
+    sudo -u postgres psql -c 'ALTER USER traffic_stops_staging WITH NOSUPERUSER;'
+
+After importing new state data into the database used by a running server,
+cached queries will continue to be used until they expire.  To flush the
+cache, connect to ``memcached`` using ``telnet`` or some other suitable
+client and send the ``flush_all`` command.
+
+Raw NC Data
+___________
+
+Command-line
+++++++++++++
 
 Run the import command:
 
 .. code-block:: bash
 
     sudo su - traffic_stops
-    cd /var/www/traffic-stops
+    cd /var/www/traffic_stops
     source ./env/bin/activate
     ./manage.sh import_nc --dest=/var/www/traffic_stops/data
 
-When finished, revoke SUPERUSER privileges:
+Reusing an existing ``--dest`` directory will speed up import.  However,
+if import code has changed since the last time the directory was used, don't
+reuse an existing directory.
 
-.. code-block:: bash
+Admin
++++++
 
-    sudo -u postgres psql -c 'ALTER USER traffic_stops_staging WITH NOSUPERUSER;'
+Follow the "Admin" instructions above under "Local/Development Environment".
 
```

Comment:
Maybe worth adding a hint about what to enter as source and destination in the admin.